packages/compass-aggregations/src/components/focus-mode/focus-mode-stage-preview.tsx

```diff
@@ -68,6 +83,61 @@ export const FocusModePreview = ({
   const docCount = documents?.length ?? 0;
   const docText = docCount === 1 ? 'document' : 'documents';
   const shouldShowCount = !isLoading && docCount > 0;
+
+  const isPipelineOptionsMenuHidden = stageOperator === '$out'
+    || stageOperator === '$merge'
+    || isMissingAtlasOnlyStageSupport;
+
+  let content = null;
+
+  if (stageOperator === '$out') {
+    content = (
+      <div className={centerStyles}>
+        <OutStagePreivew index={stageIndex} />
+      </div>
+    );
+  } else if (stageOperator === '$merge') {
+    content = (
+      <div className={centerStyles}>
+        <MergeStagePreivew index={stageIndex} />
+      </div>
+    );
+  } else if (isMissingAtlasOnlyStageSupport) {
+    content = (
+      <div className={centerStyles}>
+        <AtlasStagePreview stageOperator={stageOperator ?? ''} />
+      </div>
+    );
+  } else if (isLoading) {
+    content = (
+      <div className={centerStyles}>
+        <div className={loaderStyles}>
+          <SpinLoader title='Loading' />
+          Loading Preview Documents... 
+        </div>
+      </div>
+    );
+  } else if (documents && documents.length > 0) {
+    content = (
+      <DocumentListView
+        docs={documents.map((doc) => new HadronDocument(doc))}
```

Comment:
This will recreate documents on every render which will cause heavy rerendering of the document cards, this is also not required, you can just pass documents down to the DocumentListView, it will handle creating HadronDocument instances and memoizing them on its own

Reply:
Added changes in [becfaa0](https://github.com/mongodb-js/compass/pull/3989/commits/becfaa09b8a8a6b94d0b3c07d64c1ec3b3a20160)